Sources/TuistDependencies/SwiftPackageManager/SwiftPackageManagerInteractor.swift

```diff
@@ -53,41 +60,52 @@ public protocol SwiftPackageManagerInteracting {
 
 public final class SwiftPackageManagerInteractor: SwiftPackageManagerInteracting {
     private let fileHandler: FileHandling
+    private let swiftPackageManager: SwiftPackageManaging
 
     public init(
-        fileHandler: FileHandling = FileHandler.shared
+        fileHandler: FileHandling = FileHandler.shared,
+        swiftPackageManager: SwiftPackageManaging = SwiftPackageManager()
     ) {
         self.fileHandler = fileHandler
+        self.swiftPackageManager = swiftPackageManager
     }
 
     public func fetch(
         dependenciesDirectory: AbsolutePath,
-        dependencies: SwiftPackageManagerDependencies,
-        platforms _: Set<Platform>
+        dependencies: SwiftPackageManagerDependencies
     ) throws {
         logger.warning("Support for Swift Package Manager dependencies is currently being worked on and is not ready to be used yet.")
 
         logger.info("Resolving and fetching Swift Package Manager dependencies.", metadata: .section)
 
-        try fileHandler.inTemporaryDirectory { temporaryDirectoryPath in
-            // prepare paths
-            let pathsProvider = SwiftPackageManagerPathsProvider(
-                dependenciesDirectory: dependenciesDirectory,
-                temporaryDirectoryPath: temporaryDirectoryPath
-            )
+        try install(
+            dependenciesDirectory: dependenciesDirectory,
+            dependencies: dependencies,
+            installationMethod: { path in
+                try swiftPackageManager.resolve(at: path)
+            }
+        )
 
-            // prepare for installation
-            try loadDependencies(pathsProvider: pathsProvider, packageManifestContent: dependencies.manifestValue())
+        logger.info("Packages resolved and fetched successfully.", metadata: .subsection)
+    }
 
-            // run `Swift Package Manager`
-            let command = ["swift", "package", "--package-path", "\(temporaryDirectoryPath.pathString)", "resolve"]
-            try System.shared.runAndPrint(command)
+    public func update(
+        dependenciesDirectory: AbsolutePath,
+        dependencies: SwiftPackageManagerDependencies
+    ) throws {
+        logger.warning("Support for Swift Package Manager dependencies is currently being worked on and is not ready to be used yet.")
 
-            // post installation
-            try saveDepedencies(pathsProvider: pathsProvider)
-        }
+        logger.info("Updating Swift Package Manager dependencies.", metadata: .section)
 
-        logger.info("Packages resolved and fetched successfully.", metadata: .subsection)
+        try install(
+            dependenciesDirectory: dependenciesDirectory,
+            dependencies: dependencies,
+            installationMethod: { path in
```

Comment:
I feel like `installationMethod` as a closure brings unnecessary indirection. In this case it's IMHO fine to just pass flag such as `shouldUpdate: Bool` and do the logic inside the `install` function instead. The same for Carthage that is quite similar actually.